app/team.go

```diff
@@ -1043,14 +1043,14 @@ func (a *App) postRemoveFromTeamMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) InviteNewUsersToTeam(emailList []string, teamId, senderId string) *model.AppError {
+func (a *App) prepareInviteNewUsersToTeam(emailList []string, teamId, senderId string) (*model.User, *model.Team, *model.AppError) {
 	if !*a.Config().ServiceSettings.EnableEmailInvitations {
-		return model.NewAppError("InviteNewUsersToTeam", "api.team.invite_members.disabled.app_error", nil, "", http.StatusNotImplemented)
+		return nil, nil, model.NewAppError("InviteNewUsersToTeam", "api.team.invite_members.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
 	if len(emailList) == 0 {
```

Comment:
I think this if doesn't belong to this function (this is giving this function mixed responsibilities), looks like is there because was there before, but makes sense to check it as part of the caller function.

If we remove this, we no longer need the `emailList` here, which makes no sense for the purpose of the function.